core/jvm/src/test/scala/fs2/CompressSpec.scala

```diff
@@ -73,93 +85,426 @@ class CompressSpec extends Fs2Spec {
+          strategy: Int,
+          epochSeconds: Int,
+          fileName: String,
+          comment: String
+      ) =>
+        var (expectedFileName, actualFileName) =
+          (Option(toEncodableFileName(fileName)), Option.empty[String])
+        var (expectedComment, actualComment) =
+          (Option(toEncodableComment(comment)), Option.empty[String])
+        var (expectedMTime, actualMTime) =
+          (Option(Instant.ofEpochSecond(epochSeconds)), Option.empty[Instant])
+        Stream
+          .chunk(Chunk.bytes(s.getBytes))
+          .rechunkRandomlyWithSeed(0.1, 2)(System.nanoTime())
+          .through(
+            compress.gzip[IO](
+              509,
+              deflateLevel = Some(level),
+              deflateStrategy = Some(strategy),
+              modificationTime = Some(Instant.ofEpochSecond(epochSeconds)),
+              fileName = Some(fileName),
+              comment = Some(comment)
+            )
+          )
+          .rechunkRandomlyWithSeed(0.1, 2)(System.nanoTime())
+          .through(
+            compress.gunzip[IO](
+              1031,
+              fileNameCallback = (s: String) => { actualFileName = Some(s) },
+              commentCallback = (s: String) => { actualComment = Some(s) },
+              modificationTimeCallback = (i: Instant) => { actualMTime = Some(i) }
+            )
+          )
           .compile
           .toVector
-          .asserting(_ shouldBe s.toVector)
+          .asserting { bytes =>
+            actualFileName shouldBe expectedFileName
+            actualComment shouldBe expectedComment
+            if (epochSeconds != 0) actualMTime shouldBe expectedMTime
+            bytes shouldBe s.getBytes
+          }
     }
 
-    "gzip |> GZIPInputStream ~= id" in forAll { s: Stream[Pure, Byte] =>
-      val bytes = s
-        .through(compress.gzip(1024))
-        .compile
-        .to(Array)
+    "gzip |> GZIPInputStream ~= id" in forAll(
+      strings,
+      intsBetween(0, 9),
+      intsBetween(0, 2),
+      intsBetween(0, Int.MaxValue),
+      strings,
+      strings
+    ) {
+      (
+          s: String,
+          level: Int,
+          strategy: Int,
+          epochSeconds: Int,
+          fileName: String,
+          comment: String
+      ) =>
+        val bytes = Stream
+          .chunk(Chunk.bytes(s.getBytes))
+          .rechunkRandomlyWithSeed(0.1, 2)(System.nanoTime())
+          .through(
+            compress.gzip(
+              1024,
+              deflateLevel = Some(level),
+              deflateStrategy = Some(strategy),
+              modificationTime = Some(Instant.ofEpochSecond(epochSeconds)),
+              fileName = Some(fileName),
+              comment = Some(comment)
+            )
+          )
+          .compile
+          .to(Array)
 
-      val bis = new ByteArrayInputStream(bytes)
-      val gzis = new GZIPInputStream(bis)
+        val bis = new ByteArrayInputStream(bytes)
+        val gzis = new GZIPInputStream(bis)
```

Comment:
Did scalafmt really double-indent this?

Reply:
Yes, scalafmt made the change and it looks correct upon inspection.